### PR TITLE
Prevent unexisting aliases

### PR DESCRIPTION
**Pull request by mfolkeseth**
Fixes https://github.com/eik-lib/issues/issues/10

### CODE REVIEW

**Comment by trygve-lie** [comment]
Great. Thanks for this :)

There is a fair bit of lint adjustments here, can you give a short description of what the changes are? Also for the sake of the release notes.

**Comment by mfolkeseth** [comment]
That is correct. It does a check for package version before the alias is updated 👍

**Comment by trygve-lie** [comment]
I guess this adds a 400 - BadRequest to the alias POST documentation: https://eik.dev/docs/server_rest_api#update-alias

File: lib/handlers/alias.post.js [hunk 1]
```diff
@@ -62,47 +65,76 @@ const AliasPost = class AliasPost {
...
+                            throw new HttpError.BadRequest();
+                        }
+                        const alias = new Alias(incoming);
+                        alias.version = obj.value;
+                        return alias;
+                    }
+
+                    throw new HttpError.BadRequest();
+                })
+                .then(async (alias) => {
```
[the remark this answers is not in the document]
Comment (trygve-lie) [reply lib/handlers/alias.post.js:89]: :+1:

**Comment by mfolkeseth** [comment]
It leaves a `throw new HttpError.NotFound();` - 404 error to follow the flow of the cli https://github.com/eik-lib/cli/blob/master/classes/alias.js#L101. In which case no doc update is needed. IMO the API and CLI feedback is just fine, just missing this additional package version check.

Yes, I will absolutely add some tests. I also think it would be beneficial to pull back the lint fix in this PR to keep it more clean.

**Comment by trygve-lie** [comment]
> It leaves a throw new HttpError.NotFound(); - 404 error to follow the flow of the cli https://github.com/eik-lib/cli/blob/master/classes/alias.js#L101. In which case no doc update is needed. IMO the API and CLI feedback is just fine, just missing this additional package version check.

The API and CLI seems to be fine but there is a new status code in the POST method not reflected in the doc but I can check and update the doc when this is merged :)

> Yes, I will absolutely add some tests. 

Seems like there is just two tests failing due to the change in the returned http status code so it should be to just alter the status code in the tests. 

If your able to add a test for the version check its great :+1: 

> I also think it would be beneficial to pull back the lint fix in this PR to keep it more clean.

That would be great but might break the build. If it does its no stress. I see whats done here.

**Pull request merged**